### DISCUSSION TITLE
Execute async messages after operations

### DISCUSSION THREAD

**Discussion by AurelienFT** [massalabs/massa#3276] [Ideas]
## Context

Currently the async messages are executed before the operations without real usages. Running them after the operations could have benefits

## Benefits

- With the implementation of https://github.com/massalabs/massa/issues/3226 we could execute the messages that have their filter validate directly in the same slot.
- We could reuse the gas left after the execution of the operations to give it to the execution of messages and so allow to execute more messages if there is gas left in operations execution. If the slot didn't had a block then all the gas would go for mesage execution

**Comment by qdrn** [comment]
I don't see the difference. It's mostly an implementation detail. Executing before slot `n` is the same as executing after slot `n-1`. Maybe I'm missing something ?

**Reply by AurelienFT** [reply]
It's not the same it allow to run messages that are triggered on a condition 0.5 sec before but yeah it's a detail. By opening this discussion, I just wanted to know if there is no others problems if we change this.
